### ISSUE TITLE
Request a task from another user

### ISSUE THREAD

**Issue by levb** [mattermost/mattermost-plugin-todo#12]
Sender:
- `/todo request @user some-task`
- the task is added to the sender's todo list in the new "Sent" section; its status is displayed (pending/active). Maybe: display the position (number) in the recipient's "Requests Received" list -- this can be labelled "Priority": number, or "Received", "Declined", "Enqueued"
- When the recipient acts on the task, the sender is notified with a DM from the bot, and the task is updated (moved to "Completed") or removed from the list

Recipient:
- receives a DM from the Bot, notifying her of the new task, with action buttons.
- The task is displayed in the recipient's "Inbox" section, with applicable actions, "╳ (decline)", "🐢 (enqueue)", "✓ Done"

**Comment by levb** [comment]
cc @aaronrothschild for comments

**Comment by aaronrothschild** [comment]
> cc @aaronrothschild for comments

I love the suggestion!  Nothing comes to mind right now, but will think about this topic.

**Comment by larkox** [comment]
May I take this one?

**Comment by cpoile** [comment]
@larkox Great, go to it! :)

**Comment by larkox** [comment]
Here is a mock-up for the front-end. It would be nice to confirm if this is the idea we have in mind before starting to program it down. Keep in mind that this is a mock-up, and most of the shapes and colours are just placeholders. Feel free to propose improvements on that part.

Since there are two new sections (inbox and sent) I named the natural one "own", as in "my own todos".

This section will change slightly. Now, instead of just a button to delete the todos, we will have more accessible buttons for "Done" and "Delete". This is because some Todos will be Todos sent by other users, and we have to make the distinction whether we have done them or we just delete them.

Also, the "Created" changes to "Received from" in case the Todo is received from other user.

![image](https://user-images.githubusercontent.com/1933730/73436528-f842f500-434a-11ea-98db-a55ba7999b7d.png)


On inbox, so far I have used buttons with text. Should we use both text and images ("✓ Done" vs. "Done") only text, or only images? I would say both is better. Do we have any image we want to use? We could try to use font awesome icons like:
- Check for accept: https://fontawesome.com/icons/check?style=solid
- Cross for decline: https://fontawesome.com/icons/times?style=solid
- Plus for enqueue: https://fontawesome.com/icons/plus?style=solid

I also found nice to put the sender along the timestamp, changing "Created" by "Received".

Finally we would remove the X from the items, since we want them to decline the item, not just remove them.
![image](https://user-images.githubusercontent.com/1933730/73434153-7cdf4480-4346-11ea-8bba-eed2ccba37b6.png)

Finally, the sent section. Here the appearance of the Todo changes depending on the status.

First would be a pending status (the todo has been sent, but the receiver has not yet answered). Here we will have one button to cancel the todo, and another one (that I am not sure if would be good) "Remind" button. Remind button will send a message to the user reminding him about that single task. This can be annoying and being abused, and therefore I am not sure if it is a good feature or not.

Second status would be Enqueued. Here only remind button would be necessary. Not sure whether it makes sense to put a delete button. If a user has already enqueued the todo, he might be working already on it, so it is better to go to him and tell him to delete it on his side than deleting myself on my side. Nevertheless it is open to debate.

For enqueued and pending we add the position in the respective list: own and inbox. We consider that an accepted task becomes an owned task.

Third status would be Cancelled. This may occur after pending, or after enqueued. We keep the task there so the user can check it again. It has only one button to remove. In order to keep everything more uniform, we use a button instead of the X we used until now. In order to differentiate it and bring more attention to them, we will put some red. Right now in the mock-up is red background, but it is open to discussion.

Finally, the last status would be Done. As with cancelled, it can only be removed, and has a green background.

Apart of the status, we change the "Created" for "Sent to". Also, we add the status. Right now it is a line after the timestamp, but other solutions would be a prefix to the todo ( [PENDING] Todo text... ) or a representative icon.

![image](https://user-images.githubusercontent.com/1933730/73436253-7783f900-434a-11ea-9f83-e3b75bf1d30d.png)


The top bar should show how many "important" items are in each list.
On Inbox will be all of them. Once we mark a item as done or declined it disappears, and when we mark them as Enqueued they move to Own list.
On Sent, only those completed or deleted will be marked. Pending or Enqueued elements are not considered "important".
On Own, all items are marked.
![image](https://user-images.githubusercontent.com/1933730/73436986-ce3e0280-434b-11ea-809d-0f02e32d1b19.png)



Finally, when we add a new todo, we should have a new option: Send to user.
![image](https://user-images.githubusercontent.com/1933730/73437351-7c49ac80-434c-11ea-8319-aac145c54726.png)

What do you think? While we decide this, I will be working in the back-end side, and moving forward with the things that will probably not change.
